server/bin/utils/run-unittest

```diff
@@ -0,0 +1,607 @@
+        fi
+    fi
+
+    # Setup per-test Dataset DB
+    local _state_db=${_tdir}/state/${1}.db
+
+    # Determine which users the test needs and create them. We discover
+    # users dynamically from the database state ".db" files, but also
+    # always allow use of the satellite user for the satellite intake tests,
+    # as well as "pbench" for un-owned prep-shim intake.
+    local users="pbench satellite"
+    if [[ -f ${_state_db} ]]; then
+        users="$(cat ${_state_db} | cut --d ' ' -f 4 | sort -u) ${users}"
+    fi
+    local u
+    for u in ${users}; do
+        _create_user ${u}
+    done
+
+    if [[ -f ${_state_db} ]]; then
+        local controller
+        local name
+        local state
+        local user
+        while read -r controller name state user; do
+            pbench-state-manager --create=${user} --controller=${controller} --name=${name} --state=${state}
+        done < ${_state_db}
+    fi
+}
+
+function _reset_state {
+    # Run per-test state reset
+    local _state_reset=${_tdir}/state/${1}.reset
+    if [[ -f ${_state_reset} ]]; then
+        (cd ${_testroot}; ${_state_reset})
+        if [[ ${?} -gt 0 ]]; then
+            echo "ERROR: unable to run per-test state reset for ${1}" >&2
+            exit 1
+        fi
+    fi
+
+    export PATH=${_orig_PATH}
+    export PYTHONPATH=${_orig_PYTHONPATH}
+    unset _PBENCH_SERVER_CONFIG
+    unset SATCONFIG
+
+    rm -f ${_testopt}/lib/config/state-pbench-server.cfg
+    rm -f ${_testactout}
+    rm -f ${_testlog}
+    rm -f ${_testactlog}
+    rm -f ${_testcurlpayload}
+    rm -f ${_testloggerpayload}
+    rm -rf ${_testdir}
+    if [[ -d ${_testdir} ]]; then
+        echo "ERROR: unable to remove pbench hierarchy" >&2
+        exit 1
+    fi
+    rm -rf ${_testdir_local}
+    if [[ -d ${_testdir_local} ]]; then
+        echo "ERROR: unable to remove pbench-local hierarchy" >&2
+        exit 1
+    fi
+    rm -rf ${_testdir_sat}
+    if [[ -d ${_testdir_sat} ]]; then
+        echo "ERROR: unable to remove pbench-satellite hierarchy" >&2
+        exit 1
+    fi
+    rm -rf ${_testdir_sat_local}
+    if [[ -d ${_testdir_sat_local} ]]; then
+        echo "ERROR: unable to remove pbench-satellite-local hierarchy" >&2
+        exit 1
+    fi
+    rm -rf ${_testtmp}
+    if [[ -d ${_testtmp} ]]; then
+        echo "ERROR: unable to remove tmp hierarchy" >&2
+        exit 1
+    fi
+    rm -rf ${_testhtml}
+    if [[ -d ${_testhtml} ]]; then
+        echo "ERROR: unable to remove var-www-html hierarchy" >&2
+        exit 1
+    fi
+    rm -rf ${_testhtml_sat}
+    if [[ -d ${_testhtml_sat} ]]; then
+        echo "ERROR: unable to remove var-www-html hierarchy" >&2
+        exit 1
+    fi
+    rm -rf ${_testroot}/opt
+    if [[ -d ${_testroot}/opt ]]; then
+        echo "ERROR: unable to remove opt hierarchy" >&2
+        exit 1
+    fi
+}
+
+function _run_test {
+    # What it takes to run one test
+    local testname=${1}
+    local cmd=${2}
+    shift 2
```

Comment:
As noted elsewhere, the `shift 2` is potentially problematic...should we replace it with `shift; shift`?

Reply:
This is case it is not because `server/bin/unittests` always/must invoke `run-unittest` with the test name and the command to run, the additional arguments are optional.